gobblin-runtime/src/main/java/gobblin/runtime/app/ServiceBasedAppLauncher.java

```diff
@@ -224,13 +226,30 @@ private void addJobExecutionServerAndAdminUI(Properties properties) {
 
       if (adminUiServerEnabled) {
         LOG.info("Will launch the admin UI server");
-        addService(new AdminWebServer(properties, executionInfoServer.getAdvertisedServerUri()));
+        addService(createAdminServer(properties, executionInfoServer.getAdvertisedServerUri()));
       }
     } else if (adminUiServerEnabled) {
       LOG.warn("Not launching the admin UI because the job execution info server is not enabled");
     }
   }
 
+  public static Service createAdminServer(Properties properties,
+                                          URI executionInfoServerURI) {
+    String factoryClassName = properties.getProperty(ConfigurationKeys.ADMIN_SERVER_FACTORY_CLASS_KEY,
```

Comment:
I can explain more if you are interested, but here is the doc for now: http://docs.spring.io/spring-boot/docs/current/reference/html/using-boot-auto-configuration.html